### ISSUE TITLE
Does Action Scheduler depend on WP-Cron to execute every Minute?

### ISSUE THREAD

**Issue by balachandar-git** [woocommerce/action-scheduler#554]
I am running WP_CRON at every 6 hours. And trying to understand that how the Action Scheduler works:
Question:
Will the Action Scheduler get initiated every 6 hours (meaning at time of my WP_CRON runs)?

Thank you in advance,
Bala

**Comment by rrennick** [comment]
> Does Action Scheduler depend on WP-Cron to execute every Minute?

By default, yes. You can create a Server cron job to run actions via WP CLI on whatever schedule you want.

> Will the Action Scheduler get initiated every 6 hours (meaning at time of my WP_CRON runs)?

WP Cron only runs one scheduled event per run so if you only run it once every 6 hours it might be several days between runs (depending on how many other events are scheduled).

**Comment by balachandar-git** [comment]
Thank you a lot for quick response @rrennick.

My understanding is - let say example if WP Cron runs 6 am then the Action Scheduler will get triggered and the Action Scheduler will consider past 6 hours scheduled orders and run in a single batch.
And next 12 pm when next time WP Cron runs then the Action Scheduler will get triggered and it will consider past 6 hours scheduled orders(from 6 am to 12 pm) and run in a single batch.

Please correct me if my understanding is wrong.

One more clarification on the same -  My website has Woocomerce subscription based products and it will go through Action Scheduler. 

Please suggest me what would be the best cron time for WP Cron (considering multiple renewal payments per day)? (like */5 * * * * every five Minute 

Thank you,
Bala

**Comment by rrennick** [comment]
> My understanding is - let say example if WP Cron runs 6 am then the Action Scheduler will get triggered 

Both WordPress and WooCommerce have a number of tasks scheduled with WP Cron so AS will only be triggered if it is the task with the oldest scheduled time.

> Action Scheduler will consider past 6 hours scheduled orders and run it in a single batch.

AS does start with the oldest schedule tasks. Within the context of WP cron it runs pending tasks (scheduled in the past) up to the time limit (default 30 seconds).

> Please suggest me what would be the best cron time for WP Cron

Is there a reason you're not running it every minute?

**Comment by balachandar-git** [comment]
> Is there a reason you're not running it every minute?
Yes, if we run WP Cron every minute then our Admin Dashboard is keep getting 504 error.

**Comment by rrennick** [comment]
@balachandar-git In that case, it would probably be better to ask your hosting support for a recommended schedule.

**Comment by rrennick** [comment]
@balachandar-git I'm closing this due to lack of activity. Let us know if you have any further questions.

**Comment by mtx-z** [comment]
@rrennick Hello,

I have quite the same question: 
- Do I need WP cron to trigger the action scheduler (so doing `wp cron event run --due-now` to trigger the same as `wp action-scheduler run` would? 
- If I use `wp action-scheduler run` in my cron, do I also need the `wp cron event run` in my crontab? 
- at the moment I have both in my crontab and I don't know if thats the way to do it. 
Thanks

**Comment by rrennick** [comment]
> If I use wp action-scheduler run in my cron, do I also need the wp cron event run in my crontab?

Action Scheduler will not run other cron events. If you you want your other wp cron events to be run by server cron then you need to use the `wp cron` command. 

For AS the difference between WP CLI and WP cron is that the CLI command will continue to run until all pending actions are processed while the AS cron runner processes a limited number of actions per cron schedule.

If you do decide to schedule both cron jobs then [disable the default runner](https://github.com/woocommerce/action-scheduler-disable-default-runner) to prevent the two processes from conflicting.

**Comment by mtx-z** [comment]
@rrennick thanks a lot, it seems I missed this plugin. 

So let me resume :
- If I want to use wp-cli: I need both wp-cron and action-scheduler triggered from cron
- in this situation, I can use this plugin to disable the trigger of action-scheduler from wp-cron

Also, I'm wondering how it integrates with other plugins? Like how/why Woocommerce is adding action to the scheduler directly after I installed it? How woocommece handle those tasks before I added the scheduler plugin? If I disable action-scheduler, it means that Woocommerce won't be able to trigger its tasks? Or does it detect action-scheduler to use it instead of classic WP cron? 

Thank you again
